src/main/java/homework_2/random_chars_table/RandomCharsTable.java

```diff
@@ -0,0 +1,84 @@
+package homework_2.random_chars_table;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.InputMismatchException;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class RandomCharsTable {
+
+    private static void randomCharsTable(int lengthOfTable, int widthOfTable, String strategy) {
+
+        char[][] randomCharsTable = new char[lengthOfTable][widthOfTable];
+
+        int minRand = 'A';
+        int maxRand = 'Z';
+        for (int i = 0; i < lengthOfTable; i++) {
+            for (int j = 0; j < widthOfTable; j++) {
+                randomCharsTable[i][j] = (char) ((int) (Math.random() * (maxRand - minRand + 1) + minRand));
+                System.out.print(" | " + randomCharsTable[i][j] + " |");
+            }
+            System.out.println();
+        }
+
+        if ("even".equals(strategy.toLowerCase())) {
+            System.out.print("Even letters: ");
+        }
+
+        if ("odd".equals(strategy.toLowerCase())) {
+            System.out.print("Odd letters: ");
+        }
+
+        StringBuffer stringBuffer = new StringBuffer();
+
+        for (int i = 0; i < lengthOfTable; i++) {
+            for (int j = 0; j < widthOfTable; j++) {
+                int codeOfChar = randomCharsTable[i][j];
+                if ("even".equals(strategy.toLowerCase())) {
+                    if (codeOfChar % 2 == 0) {
+                        stringBuffer.append(" ").append((char) codeOfChar).append(" ,");
+                    }
+                }
+
+                if ("odd".equals(strategy.toLowerCase())) {
+                    if (codeOfChar % 2 != 0) {
+                        stringBuffer.append(" ").append((char) codeOfChar).append(" ,");
+                    }
+                }
+            }
+        }
+
+        stringBuffer.deleteCharAt(stringBuffer.length() - 1);
```

Comment:
Execution failed for task ':Main.main()'.
what if your 'stringBuffer' is empty? try to run with "1 1 even" or "1 1 odd" as input

Reply:
oh thank you! let me check